core/src/main/kotlin/gateway/handler/InteractionEventHandler.kt

```diff
@@ -0,0 +1,63 @@
+package dev.kord.core.gateway.handler
+
+import com.gitlab.kordlib.cache.api.DataCache
+import dev.kord.core.Kord
+import dev.kord.core.cache.data.ApplicationCommandData
+import dev.kord.core.cache.data.InteractionData
+import dev.kord.core.entity.GuildApplicationCommand
+import dev.kord.core.entity.interaction.Interaction
+import dev.kord.core.entity.interaction.PartialInteraction
+import dev.kord.core.event.interaction.*
+import dev.kord.core.gateway.MasterGateway
+import dev.kord.gateway.*
+import kotlinx.coroutines.flow.MutableSharedFlow
+import dev.kord.core.event.Event as CoreEvent
+
+class InteractionEventHandler(
+    kord: Kord,
+    gateway: MasterGateway,
+    cache: DataCache,
+    coreFlow: MutableSharedFlow<CoreEvent>
+) : BaseGatewayEventHandler(kord, gateway, cache, coreFlow) {
+    override suspend fun handle(event: Event, shard: Int) {
+        when(event) {
+            is InteractionCreate -> handle(event, shard)
+            is ApplicationCommandCreate -> handle(event, shard)
+            is ApplicationCommandUpdate -> handle(event, shard)
+            is ApplicationCommandDelete -> handle(event, shard)
+            else -> Unit
+        }
+
+    }
+
+    private suspend fun handle(event: InteractionCreate, shard: Int) {
+
+        val data = InteractionData.from(event)
+        val partial = PartialInteraction(data, kord)
+        coreFlow.emit(PartialInteractionCreateEvent(partial, kord, shard))
+
+        if (kord.resources.applicationId != null) {
+            val interaction = Interaction(data, kord.resources.applicationId, kord)
+            coreFlow.emit(InteractionCreateEvent(interaction, kord, shard))
+        }
```

Comment:
This will result in the same event getting sent twice if app id is present. Consider only the most relevant one instead.